server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/SubAddressing.java

```diff
@@ -55,20 +63,36 @@ public void service(Mail mail) throws MessagingException {
                 .ifPresent(
                     Throwing.consumer(details -> {
 
-                        boolean hasRight = mailboxManager.hasRight(
-                            MailboxPath.forUser(
-                                Username.fromMailAddress(recipient.stripDetails("+")),
-                                details
-                            ),
-                            MailboxACL.Right.Post,
-                            session
-                        );
+                        boolean hasRight;
+
+                        try {
+                            Username recipientUsername = Username.fromLocalPartWithoutDomain(recipient.getLocalPartWithoutDetails("+"));
+                            MailboxSession session = mailboxManager.createSystemSession(recipientUsername);
 
-                        String targetFolder = hasRight ? details : "inbox";
+                            MailboxACL aclCommand = mailboxManager.getMailbox(
+                                    MailboxPath.forUser(recipientUsername, details), session)
+                                .getMetaData(IGNORE, session, MessageManager.MailboxMetaData.FetchGroup.NO_COUNT)
+                                .getACL();
 
+                            hasRight = aclCommand.getEntries().containsKey(MailboxACL.ANYBODY_KEY) &&
+                                aclCommand.getEntries().get(MailboxACL.ANYBODY_KEY).equals(MailboxACL.Rfc4314Rights.of(Collections.singletonList(MailboxACL.Right.Post)));
+
+                        } catch (MailboxNotFoundException e) {
+                            hasRight = false;
+                        }
+
+                        String targetFolder;
+                        if(hasRight) {
+                            targetFolder = details;
+                        } else {
+                            // TODO find LOG
+                            // LOG.info(maybesender + " tried to address " + recipient + "'s subfolder `" + details + "` but did not have the right to")
+                            targetFolder = MailboxConstants.INBOX;
```

Comment:
Nope else we just do not do something. IE we only put the storage directive when we have rights?